cpp/tests/fixed_point/fixed_point_tests.cu

```diff
@@ -0,0 +1,453 @@
+
+    using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
+
+    std::vector<decimal32> vec1(size);
+    std::vector<ValueType> vec2(size);
+
+    std::iota(std::begin(vec1), std::end(vec1), decimal32{initial_value, scale_type{scale}});
+    std::iota(std::begin(vec2), std::end(vec2), initial_value);
+
+    auto const res1 = std::accumulate(
+        std::cbegin(vec1),
+        std::cend(vec1),
+        decimal32{0, scale_type{scale}});
+
+    auto const res2 = std::accumulate(
+        std::cbegin(vec2),
+        std::cend(vec2),
+        static_cast<ValueType>(0));
+
+    EXPECT_EQ(static_cast<int32_t>(res1), res2);
+
+    std::vector<ValueType> vec3(vec1.size());
+
+    std::transform(
+        std::cbegin(vec1),
+        std::cend(vec1),
+        std::begin(vec3),
+        [] (auto const& e) { return static_cast<int32_t>(e); });
+
+    EXPECT_EQ(vec2, vec3);
+}
+
+TEST_F(FixedPointTest, Decimal32IntVector) {
+
+    integer_vector_test(0,   10,   -2, std::plus<>());
+    integer_vector_test(0,   1000, -2, std::plus<>());
+
+    integer_vector_test(1, 10, 0, std::multiplies<>());
+    integer_vector_test(2, 20, 0, std::multiplies<>());
+
+}
+
+template<typename ValueType, typename Binop>
+void float_vector_test(ValueType const initial_value,
+                       int32_t   const size,
+                       int32_t   const scale, Binop binop) {
+
+    using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
+
+    std::vector<decimal32> vec1(size);
+    std::vector<ValueType> vec2(size);
+
+    std::iota(std::begin(vec1), std::end(vec1), decimal32{initial_value, scale_type{scale}});
+    std::iota(std::begin(vec2), std::end(vec2), initial_value);
+
+    auto equal = std::equal(
+        std::cbegin(vec1),
+        std::cend(vec1),
+        std::cbegin(vec2),
+        [] (auto const& a, auto const& b) {
+            return static_cast<double>(a) - b <= std::numeric_limits<ValueType>::epsilon();
+        });
+
+    EXPECT_TRUE(equal);
+}
+
+TEST_F(FixedPointTest, Decimal32FloatVector) {
+
+    float_vector_test(0.1,  1000, -2, std::plus<>());
+    float_vector_test(0.15, 1000, -2, std::plus<>());
+
+    float_vector_test(0.1,  10, -2, std::multiplies<>());
+    float_vector_test(0.15, 20, -2, std::multiplies<>());
+
+}
+
+struct cast_to_int32_fn
+{
+    using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
+    int32_t __host__ __device__ operator()(decimal32 fp) {
+        return static_cast<int32_t>(fp);
+    }
+};
+
+TEST_F(FixedPointTest, DecimalXXThrustOnDevice) {
+
+    using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
+
+    thrust::device_vector<decimal32> vec1(1000, decimal32{1, scale_type{-2}});
+
+    auto const sum = thrust::reduce(
+        rmm::exec_policy(0)->on(0),
+        std::cbegin(vec1),
+        std::cend(vec1),
+        decimal32{0, scale_type{-2}});
+
+    EXPECT_EQ(static_cast<int32_t>(sum), 1000);
+
+    // thrust::sequence doesn't work on device without modifying fixed_point
```

Comment:
Modifying in what way? We need it to work in any thrust algorithm, right?

Reply:
Suggest replacing this with a TODO to update to device algorithms with a link to the thrust issue.

Reply:
Done!